### PR TITLE
PartialEq between PolyOverZq and ModPoly

### PR DESCRIPTION
**Pull request by Marcel583**
**Description**

This PR implements PartialEq between PolyOverZq and ModPoly.

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/modulus_polynomial_ring_zq/cmp.rs [hunk 1]
```diff
@@ -234,3 +293,33 @@ mod test_partial_eq {
...
+        let q = ModulusPolynomialRingZq::from_str("4  1 2 3 4 mod 17").unwrap();
+        let z = PolyOverZq::from_str("4  1 2 3 4 mod 17").unwrap();
+
+        assert!(q == z);
+        assert!(z == q);
+        assert!(&q == &z);
+        assert!(&z == &q);
+    }
+
+    // Ensure that large values are compared correctly
```
Comment (jnsiemer) [inline src/integer_mod_q/modulus_polynomial_ring_zq/cmp.rs:316]:
```suggestion
    /// Ensure that large values are compared correctly.
```

File: src/integer_mod_q/modulus_polynomial_ring_zq/cmp.rs [hunk 2]
```diff
@@ -234,3 +293,33 @@ mod test_partial_eq {
         assert_ne!(first, second);
     }
 }
+
+/// Test that the [`PartialEq`] trait is correctly implemented.
+#[cfg(test)]
+mod test_partial_eq_q_other {
```
Comment (jnsiemer) [inline src/integer_mod_q/modulus_polynomial_ring_zq/cmp.rs:299]: Please add a test for a polynomial that isn't equal to the other one (same number of coefficients).
Furthermore, please add two tests for different numbers of coefficients (one that is equal with 0 as largest coefficient, one for different ones but different lengths). - To make sure that the function works for polynomials of different degrees, too.

**Review by jnsiemer — APPROVED** [review]
Looks good. Please add the tests before merging.
